pkg/queue/instance.go

```diff
@@ -99,12 +116,18 @@ func (q *queueImpl) get() (task Task, shutdown bool) {
 
 	if q.closing && len(q.tasks) == 0 {
 		// We must be shutting down.
-		return nil, true
+		return queueTask{}, true
 	}
 	task = q.tasks[0]
+	task.startTime = time.Now()
 	// Slicing will not free the underlying elements of the array, so explicitly clear them out here
-	q.tasks[0] = nil
+	q.tasks[0] = queueTask{}
```

Comment:
It is still holding memory allocated to empty queueTask.

The smooth change would be making use of queueTask pointer type

Reply:
I thought the same, when was writing. I will convert it to a pointer queue. Thanks for the suggestion!

Reply:
FWIW this line is still important since it frees the `func` in the task which may be a closure over some large variables. But a pointer would be 8bytes and queueTask is 56